pageserver/src/layered_repository/filename.rs

```diff
@@ -269,38 +306,52 @@ impl fmt::Display for ImageFileName {
     }
 }
 
+/// All files, found on the disk for the corresponding timeline.
+#[derive(Debug)]
+pub struct TimelineFiles {
+    pub image_layer_names: BTreeSet<ImageFileName>,
+    // Note that the order matters here, ergo [`BTreeSet`].
+    // The delta files are created in order starting from the oldest file,
+    // because each DeltaLayer needs a reference to its predecessor.
+    // [`DeltaFileName`] derives the ordering which is used here.
+    pub delta_layer_names: BTreeSet<DeltaFileName>,
```

Comment:
Since there's no online querying, does it make more sense to just use Vecs and sort them before returning?

Reply:
But circling back to https://github.com/zenithdb/zenith/pull/688/files#r723812773, looks like this part can stay the same, less new code is great, thanks.

Reply:
With the removal of predecessors in #717 order shouldn't matter anymore